lib/ace/autocomplete.js

```diff
@@ -434,7 +434,8 @@ var FilteredList = function(array, filterText) {
         this.filterText = str;
         matches = this.filterCompletions(matches, this.filterText);
         matches = matches.sort(function(a, b) {
-            return b.exactMatch - a.exactMatch || a.meta > b.meta || b.score - a.score;
+            return (a.exactMatch < b.exactMatch) || ((b.meta < a.meta) - (a.meta < b.meta))
```

Comment:
comparator is supposed to return integer value [[docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)]

Here you return booleans which may match the expected ints somehow due to crazy JS type conversions, but are extremely fragile and error prone. Please rewrite it so that it explicitly returned integer.